features/step_definitions/networking.rb

```diff
@@ -678,9 +678,9 @@
   ensure_admin_tagged
   hosts = step "I select a random node's host"
   cb_name = "interface" unless cb_name
-  @result = host.exec_admin("/sbin/ip route show default | awk '/default/ {print $5}'")
+  @result = host.exec_admin("/sbin/ip route show default | awk 'NR==1{print $5}'")
```

Comment:
was there bad input with multiple lines?  I'm the past I've seen `ip -o -4` to ensure single-line records and IPv4 only, but that might not be helpful in this case.

Also curiously `ip -j` outputs JSON now, which is exciting if we had `jq` installed everywhere.

```
ip -4  -j route show default | jq -r '.[0] | .dev'
```

Reply:
Great suggestion @rbbratta but jq is not installed on nodes.
/sbin/ip route show default outputs like this:
default via 10.0.128.1 dev **ens3** 
default via 10.0.128.1 dev **ens3** proto dhcp metric 100 

There can be various way to grep ens3 but thought it should take only single line and fetch ens3 from that else it would fetch ens3 multiple times which needed a fix hence this one

Reply:
`host.exec_admin` is using ssh maybe?  

oddly, there seems to be `jq`  with `oc debug` and `chroot /host`

```
sh-4.4# cat /etc/redhat-release
Red Hat Enterprise Linux CoreOS release 4.3
sh-4.4# which jq
/usr/bin/jq
sh-4.4# jq --version
jq-1.5
```

Reply:
host.exec_admin triggers oc debug. Hmm, with my oc debug on 4.3:

sh-4.4# cat /etc/redhat-release 
Red Hat Enterprise Linux CoreOS release 4.3
sh-4.4# which jq
which: no jq in (/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin)

It is recommended to depend on basic linux commands/tools as across releases the support for tools keeps changing like for jq or git etc. Happened when we transitioned from 3.x to 4.x

Reply:
exciting, must be different version of the debug pod or something.  

`    Image ID:      registry.redhat.io/rhel7/support-tools@sha256:6d0d66b55dbc1917f15b6c048158174b5bb5813bf10bb8d74c962a65dbb217dc`

Reply:
Need to look deeper in this pull request. One thing discouraged since 4.x is running commands on nodes. People usually run commands on some pod when needed.

If possible do the same. Presently we are running commands on nodes using `oc debug node`. But it has a few nasty limitations:
https://bugzilla.redhat.com/show_bug.cgi?id=1771552
https://bugzilla.redhat.com/show_bug.cgi?id=1771549
https://bugzilla.redhat.com/show_bug.cgi?id=1771553

Reply:
I am afraid that's not possible at the moment :( multus pods create their own interfaces different from node and we are interested in finding master interface on node. like multus pod create eth0 as primart interface but node create ensX. I tried to execute same and different command inside pod but couldn't get ensX in output

Reply:
@anuragthehatter  can you try execute command on sdn pod instead of nodes?
When I run command on the "<%= cb.node_name %>" node's sdn pod:
     | bash | -c | /sbin/ip route show default | awk 'NR==1{print $5}  |

Reply:
oops this is a general step for collecting master inf name. Sorry i misunderstood it as collecting master inf name from multus pod which is not possible. Yep, let me fix this to go inside pod for collecting master inf. It should be done via sdn pod or ovnkube pod if its OVN cluster.

Reply:
@akostadinov committed. PTAL. Thanks